src/main/java/picard/sam/BamErrorMetric/BaseErrorAggregation.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018 The Broad Institute
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package picard.sam.BamErrorMetric;
+
+import htsjdk.samtools.reference.SamLocusAndReferenceIterator.SAMLocusAndReference;
+import htsjdk.samtools.util.CollectionUtil;
+import htsjdk.samtools.util.SamLocusIterator;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TreeSet;
+import java.util.function.Supplier;
+
+/**
+ * An interface and implementations for classes that apply a {@link picard.sam.BamErrorMetric.ReadBaseStratification.RecordAndOffsetStratifier RecordAndOffsetStratifier}
+ * to put bases into various "bins" and then compute an {@link ErrorMetrics} on these bases using a {@link BaseErrorCalculation}.
+ *
+ * @author Yossi Farjoun
+ */
+
+public class BaseErrorAggregation<CALCULATOR extends BaseErrorCalculation.BaseCalculator> {
+    private final Supplier<CALCULATOR> simpleAggregatorGenerator;
+    private final ReadBaseStratification.RecordAndOffsetStratifier stratifier;
+
+    // Updated as a defaulting map.
+    @SuppressWarnings("MismatchedQueryAndUpdateOfCollection")
+    private final CollectionUtil.DefaultingMap<Object, CALCULATOR> strataAggregatorMap = new CollectionUtil.DefaultingMap<>
```

Comment:
Declared type should be more general, unless a more specific type is required. Using `Map` here should work. Also, the anonymous subclass can be replaced with a lambda.
```
    private final Map<Object, CALCULATOR> strataAggregatorMap = 
            new CollectionUtil.DefaultingMap<>(ignored -> simpleAggregatorGenerator.get(), true);
```